### PR TITLE
MISUV-4290 Upgrade to Scala 2.13.x - income-tax-view-change

### PR DESCRIPTION
**Pull request by andriikondiuk**

### CODE REVIEW

File: project/plugins.sbt [hunk 1]
```diff
@@ -8,10 +8,10 @@ addSbtPlugin("uk.gov.hmrc" % "sbt-auto-build" % "3.6.0")
...
-addSbtPlugin("org.scoverage" % "sbt-scoverage" % "1.8.2")
+addSbtPlugin("org.scoverage" % "sbt-scoverage" % "2.0.4")
 
 addSbtPlugin("org.scalastyle" %% "scalastyle-sbt-plugin" % "1.0.0")
 
 addSbtPlugin("ch.epfl.scala" % "sbt-scalafix" % "0.9.29")
 
-addSbtPlugin("com.typesafe.play" % "sbt-plugin" % "2.8.7")
\ No newline at end of file
+addSbtPlugin("com.typesafe.play" % "sbt-plugin" % "2.8.15")
```
Comment (avelx) [inline project/plugins.sbt:17]: just wondering if com.typesafe.play plugging upgrade is something required as a part of scala upgrade?

File: test/utils/MaterializerSupport.scala [hunk 2]
```diff
@@ -17,9 +17,7 @@
 package utils
 
 import akka.actor.ActorSystem
-import akka.stream.ActorMaterializer
```
Comment (avelx) [inline test/utils/MaterializerSupport.scala:20]: from what I can see we are still on akka 2.6.x, which is outside of scope of licensing changes they introduced recently (for more details => event-akka-licence-change channel); but we should be careful going forward as this anything above 2.7.x would required a license;

Comment (andriikondiuk) [reply test/utils/MaterializerSupport.scala:20]: Thus was needed to remove the compiler warnings

File: it/helpers/WiremockHelper.scala [hunk 3]
```diff
@@ -94,15 +94,19 @@ trait WiremockHelper {
...
+  def stopWiremock(): Unit = wireMockServer.stop()
 
-  def resetWiremock() = WireMock.reset()
+  def resetWiremock(): Unit = WireMock.reset()
 
-  def buildClient(path: String) = ws.url(s"http://localhost:$port/income-tax-view-change$path").withFollowRedirects(false)
+  def buildClient(path: String): WSRequest =
+    ws
+      .url(s"http://localhost:$port/income-tax-view-change$path")
+      .withHttpHeaders("Authorization" -> "Bearer123")
```
Comment (avelx) [inline it/helpers/WiremockHelper.scala:109]: why do we need authorisation added? I would assume this used to work in the past?

Comment (andriikondiuk) [reply it/helpers/WiremockHelper.scala:109]: It's what I was mentioned Yesterday on the Stand-up call. And I've added the explanation to the Jira ticket (https://jira.tools.tax.service.gov.uk/browse/MISUV-4251): 
If you are migrating to Scala 2.13.8 and see your Integration tests are failing with 401 then, please read the link of how to resolve it
https://hmrcdigital.slack.com/archives/C0J8BH46N/p1655305814832349

**Pull request merged**